quinn-proto/src/connection/mod.rs

```diff
@@ -432,21 +432,15 @@ where
         let mut congestion_blocked = false;
 
         // Iterate over all spaces and find data to send
-        let mut space_idx = 0;
-        let spaces = [SpaceId::Initial, SpaceId::Handshake, SpaceId::Data];
-        while space_idx < spaces.len() {
-            let space_id = spaces[space_idx];
-
+        for space_id in SpaceId::iter() {
```

Comment:
This was explicitly changed for GSO, and changing it back won't work. There's a comment at the end of the function why.

Reply:
Ah, oops. The comment wasn't very discoverable from the point of where the iteration happens, so I replaced my previous commit with a new one that explains this closer to the loop setup (8fa5c32).